src/accounts.rs

```diff
@@ -1,216 +1,336 @@
+    }
+}
+
+impl KeyPair for FreezerKeyPair {
+    type PubKey = FreezerPubKey;
+
+    fn pub_key(&self) -> Self::PubKey {
+        self.pub_key()
+    }
+}
+
+impl KeyPair for UserKeyPair {
+    // The `PubKey` here is supposed to be a conceptual "primary key" for looking up
+    // `UserKeyPairs`. We typically want to look up `UserKeyPairs` by `Address`, not `UserPubKey`,
+    // because if we have a `UserPubKey` we can always get an `Address` to do the lookup.
+    type PubKey = UserAddress;
+
+    fn pub_key(&self) -> Self::PubKey {
+        self.address()
+    }
+}
+
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[serde(bound = "Key: DeserializeOwned + Serialize")]
+pub struct Account<L: Ledger, Key: KeyPair> {
+    /// The account key.
+    key: Key,
+    /// The account description.
+    pub description: String,
+    /// Whether the account is used.
+    used: bool,
+    /// Optional ledger scan.
+    scan: Option<BackgroundKeyScan<L>>,
+    /// The time when the account was created.
+    created_time: DateTime<Local>,
+    /// The last time when the account was modified.
+    modified_time: DateTime<Local>,
+}
+
+impl<L: Ledger, Key: KeyPair> Account<L, Key> {
+    #![allow(dead_code)]
+
+    /// Get the account key.
+    pub fn key(&self) -> &Key {
+        &self.key
+    }
+
+    /// Get the account public key.
+    pub fn pub_key(&self) -> Key::PubKey {
+        self.key.pub_key()
+    }
+
+    /// Get the account description.
+    pub fn description(&self) -> String {
+        self.description.clone()
+    }
+
+    /// Check whether the account is used.
+    pub fn used(&self) -> bool {
+        self.used
+    }
+
+    /// Get the optional ledger scan.
+    pub fn scan(&self) -> Option<BackgroundKeyScan<L>> {
+        self.scan.clone()
+    }
+
+    /// Get the created time.
+    pub fn created_time(&self) -> DateTime<Local> {
+        self.created_time
+    }
+
+    /// Get the modified time.
+    pub fn modified_time(&self) -> DateTime<Local> {
+        self.modified_time
+    }
+}
+
+impl<L: Ledger, Key: KeyPair> PartialEq<Self> for Account<L, Key> {
+    fn eq(&self, other: &Self) -> bool {
+        // We assume that the private keys are equal if the public keys are.
+        self.key.pub_key() == other.key.pub_key()
+            && self.description == other.description
+            && self.used == other.used
+            && self.scan == other.scan
+            && self.created_time == other.created_time
+            && self.modified_time == other.modified_time
+    }
+}
+
+pub type AccountsStore<L, Key, PubKey> = KeyValueStore<PubKey, Account<L, Key>>;
+
+/// An editor to create or update the account or accounts store.
+pub struct AccountEditor<'a, L: Ledger, Key: KeyPair + DeserializeOwned + Serialize> {
+    account: Account<L, Key>,
+    store: &'a mut AccountsStore<L, Key, Key::PubKey>,
+}
+
+impl<'a, L: Ledger, Key: KeyPair + DeserializeOwned + Serialize> AccountEditor<'a, L, Key> {
+    #![allow(dead_code)]
```

Comment:
Still needed?

Reply:
Turned out that `with_scan` and `clear_scan` weren't used, so I removed them for now. We could just add them back if needed.

I also removed two `#![allow(dead_code)]` from `assets.rs`.